src/components/node-list/styles/_filter.scss

```diff
@@ -14,72 +39,89 @@
     fill: none;
     stroke: var(--color-nodelist-filter-indicator-off);
   }
-}
 
-// Fade row text when unchecked
-.pipeline-nodelist__row--unchecked .pipeline-nodelist__row__text--kind-filter {
-  opacity: 0.55;
+  .pipeline-nodelist__row:hover &.pipeline-row__toggle-icon--all-unset,
+  [data-whatintent='keyboard'] .pipeline-nodelist__row__checkbox:focus + &,
+  &.pipeline-row__toggle-icon--parent {
+    fill: $color-primary;
+    stroke: $color-primary;
+  }
 }
 
-// Brighter row text when unchecked and hovered
-.pipeline-nodelist__row--unchecked:hover
-  .pipeline-nodelist__row__text--kind-filter {
-  opacity: 0.8;
-}
+//--- Toggle icon opacities ---//
 
-// Bright row text when all unset
-.pipeline-nodelist__group--all-unset
-  .pipeline-nodelist__row--unchecked
-  .pipeline-nodelist__row__text--kind-filter {
-  opacity: 1;
-}
+/*
+ Parent (toggle all tags):
+ | row-hover | icon-hover | checked | opacity | ID |
+ |    🚫     |     🚫      |    🚫   |    0    | 0  |
+ |    ✅     |     🚫      |    🚫   |   0.3   | 1  |
+ |    ✅     |     ✅      |    🚫   |   0.55  | 2  |
+ |    🚫     |     🚫      |    ✅   |   0.55  | 2  |
+ |    ✅     |     ✅      |    ✅   |   0.75  | 3  |
 
-// Hide all indicators when all unset
-.pipeline-nodelist__group--all-unset .pipeline-row__toggle--kind-filter {
-  opacity: 0;
-}
+ Child (individual tag):
+ | row-hover | checked | opacity | ID |
+ |    🚫     |    🚫    |    0    | 0  |
+ |    ✅     |    🚫    |   0.3   | 1  |
+ |    🚫     |    ✅    |   0.55  | 2  |
+ |    ✅     |    ✅    |   0.75  | 3  |
+ */
 
-// Show indicator when all unset and row hovered
-.pipeline-nodelist__group--all-unset
-  .pipeline-nodelist__row:hover
-  .pipeline-row__toggle--kind-filter {
-  opacity: 0.55;
+$icon-opacity-0: 0;
+$icon-opacity-1: 0.3;
+$icon-opacity-2: 0.55;
+$icon-opacity-3: 0.75;
 
-  &:hover {
-    opacity: 0.55;
+.pipeline-row__toggle-icon--kind-filter {
```

Comment:
From my previous experience of working with these styles without prior context, one of the main things I find hard to grasp amongst our class name declarations is seperating classes for the 'parent' and 'child' toggle button. I believe this button refers to the 'parent' toggle button for tags - perhaps it is worth to name it as `.pipeline-row__toggle-icon--kind-filter-parent` to indicate this is for the parent toggle button?

Reply:
I agree, but I'm already doing that here. In the updated CSS in this PR, if it doesn't use the `--parent` modifier then it's not intended to apply to just the parent. The line you're referring to here (line 75) applies to both the parent and child icons. I've got other rules below that use `--parent` and `--child` modifiers, where I wanted to specify one or the other (to be more explicit than it was previously).

Reply:
We have four declarations in this set that are intended to apply to both parents and children (so they don't include a specifying modifier), and three that are specifically for just parents or just children.